### PR TITLE
Upgrade alpine, go, and github.com/Azure/azure-sdk-for-go/sdk/azidentity

### PR DESCRIPTION
**Pull request by dsabsay**
**What this PR does**:

* Upgrade to Alpine 3.18 to fix some security vulnerabilities
* Upgrade (patch version) go

**Which issue(s) this PR fixes**:
Fixes #<issue number>

**Checklist**
- [ ] Tests updated
- [ ] Documentation added
- [x] `CHANGELOG.md` updated - the order of entries should be `[CHANGE]`, `[FEATURE]`, `[ENHANCEMENT]`, `[BUGFIX]`

### CODE REVIEW

File: build-image/Dockerfile [hunk 1]
```diff
@@ -1,4 +1,4 @@
-FROM golang:1.21.9-bullseye
+FROM golang:1.21.11-bullseye
```
Comment (friedrichg) [inline build-image/Dockerfile:1]: Building images locally is a pain. Let's if we can merge https://github.com/cortexproject/cortex/pull/6026 before this PR and make this PR easier

Comment (friedrichg) [reply build-image/Dockerfile:1]: @dsabsay ok that was merged, make a separate PR for this Dockerfile change as explained on https://cortexmetrics.io/docs/contributing/how-to-update-the-build-image/

**Comment by dsabsay** [comment]
The failing test passed on my laptop. Can we re-run the action/workflow?

**Review by friedrichg — APPROVED** [review]
Thanks! @dsabsay

**Pull request merged**